solidity/contracts/igps/InterchainGasPaymaster.sol

```diff
@@ -132,6 +133,38 @@ contract InterchainGasPaymaster is
 
     // ============ Public Functions ============
 
+    /**
+     * @notice Deposits msg.value as a payment for the relaying of a message
+     * to its destination chain.
+     * @dev Overpayment will result in a refund of native tokens to the _refundAddress.
+     * Callers should be aware that this may present reentrancy issues.
+     * @param _messageId The ID of the message to pay for.
+     * @param _destinationDomain The domain of the message's destination chain.
+     * @param _gasAmount The amount of destination gas to pay for.
+     * @param _refundAddress The address to refund any overpayment to.
+     */
+    function payForGas(
+        bytes32 _messageId,
+        uint32 _destinationDomain,
+        uint256 _gasAmount,
+        address _refundAddress
```

Comment:
still not sure why this shows up in the diff
in future you should avoid relocating code, especially if it's unchanged to minimize the diff

Reply:
i added postDispatch and git is dumb